uulib/uucheck.c

```diff
@@ -193,7 +193,7 @@ UUGetFileName (char *subject, char *ptonum, char *ptonend)
 
   if (length == 0) {
     ptr = subject;
-    while ((iter = strchr (ptr, '/')) != NULL) {
+    while ((iter <= ptonend) && ((iter = strchr (ptr, '/')) != NULL)) {
```

Comment:
The addition of `iter <= ptonend` is not clear to me for a fix, because:
- Other parts of the while loop body may stll run `iter` over/beyond `ptonend` (if `ptonend` is meant to be a hard "no byte access after this" limit) happily
- `strchr` is inspecting `ptr` rather than `iter` and `ptr` may bee advanced even further because of code `ptr = iter + ((count)?count:1);` near the end of the while loop.

Reply:
You are right, this does not make much sense like that. What should work is to use ptr <= subject+strlen(subject) - not sure if the best way, but should do.